Worksheet_D/oxo.py

```diff
@@ -0,0 +1,115 @@
+class OxoBoard:
+    def __init__(self):
+        """ The initialiser. Initialise any fields you need here. """
+        #This command sets up the gameboard as 9 different values each beginning with 0(empty)
+        self.game_board = [0, 0, 0, 0, 0, 0, 0, 0, 0]
+
+    def get_square(self, x, y):
+        """ Return 0, 1 or 2 depending on the contents of the specified square. """
+        #Determines the location of a particular cell on the board using the equation y * 3 + x, so height of the co-ordinate input multiplied by 3 + the width of the co-ordinate input.
+        return self.game_board[y * 3 + x]
+
+    def set_square(self, x, y, mark):
+        """ If the specified square is currently empty (0), fill it with mark and return True.
+            If the square is not empty, leave it as-is and return False. """
+        #This algorithm accomplishes what the above two lines are asking for. The code checks get_square for the value 0 and if it is 0 returns with a mark.
+        if self.get_square(x, y) == 0:
+            self.game_board[y * 3 + x] = mark
+            return True
+        else:
+            return False
+
+    def is_board_full(self):
+        """ If there are still empty squares on the board, return False.
+            If there are no empty squares, return True. """
+        #Once again the 2 lines above sum this up pretty well. this code goes through the length of the game_board and if any cell is empty it will return true
+        for i in range(len(self.game_board)):
+            if self.game_board[i] == 0:
+                return False
+
+        return True
```

Comment:
Could add a comment here: "If execution reaches this point, then there are no empty squares on the board; that is, the board is full."